core/src/main/resources/error/error-classes.json

```diff
@@ -77,29 +110,73 @@
     "message" : [ "PARTITION clause cannot contain a non-partition column name: %s" ],
     "sqlState" : "42000"
   },
+  "NOT_EXPECTED_UNRESOLVED_ENCODER" : {
+    "message" : [ "Unresolved encoder expected, but %s was found." ],
+    "sqlState" : "42000"
+  },
+  "NOT_OVERRIDE_EXPECTED_METHODS" : {
+    "message" : [ "%s must override either %s or %s" ],
+    "sqlState" : "42000"
+  },
+  "NULL_AS_MAP_KEY_NOT_ALLOWED" : {
+    "message" : [ "Cannot use null as map key!" ],
+    "sqlState" : "42000"
+  },
   "PIVOT_VALUE_DATA_TYPE_MISMATCH" : {
     "message" : [ "Invalid pivot value '%s': value data type %s does not match pivot column data type %s" ],
     "sqlState" : "42000"
   },
+  "PRIMARY_CONSTRUCTOR_NOT_FOUND" : {
+    "message" : [ "Couldn't find a primary constructor on %s" ],
+    "sqlState" : "42000"
+  },
   "RENAME_SRC_PATH_NOT_FOUND" : {
     "message" : [ "Failed to rename as %s was not found" ],
     "sqlState" : "22023"
   },
+  "RESOLVE_CANNOT_HANDLE_NESTED_SCHEMA" : {
+    "message" : [ "Can not handle nested schema yet...  plan %s" ],
+    "sqlState" : "42000"
+  },
   "SECOND_FUNCTION_ARGUMENT_NOT_INTEGER" : {
     "message" : [ "The second argument of '%s' function needs to be an integer." ],
     "sqlState" : "22023"
   },
   "UNABLE_TO_ACQUIRE_MEMORY" : {
     "message" : [ "Unable to acquire %s bytes of memory, got %s" ]
   },
+  "UNEXPECTED_OPERATOR_IN_CORRELATED_SUBQUERY" : {
+    "message" : [ "Unexpected operator %s in correlated subquery%s" ],
+    "sqlState" : "42000"
+  },
+  "UNREACHABLE" : {
+    "message" : [ "This line should be unreachable%s" ],
+    "sqlState" : "42000"
+  },
   "UNRECOGNIZED_SQL_TYPE" : {
     "message" : [ "Unrecognized SQL type %s" ],
     "sqlState" : "42000"
   },
+  "UNSUPPORTED_ENCODER" : {
+    "message" : [ "Only expression encoders are supported for now." ],
```

Comment:
We should [avoid](https://spark.apache.org/error-message-guidelines.html) making promises about future support. Can you remove `for now`?